posixlib/src/main/resources/scala-native/sys/stat.c

```diff
@@ -42,9 +42,18 @@ void scalanative_stat_init(struct stat *stat,
     my_stat->st_uid = stat->st_uid;
     my_stat->st_gid = stat->st_gid;
     my_stat->st_size = stat->st_size;
-    my_stat->_st_atime = stat->st_atime;
-    my_stat->_st_mtime = stat->st_mtime;
-    my_stat->_st_ctime = stat->st_ctime;
+// see https://linux.die.net/man/2/stat
+#if defined(_BSD_SOURCE) || defined(_SVID_SOURCE) ||                           \
+    defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200809L ||                  \
+    defined(_XOPEN_SOURCE) && _XOPEN_SOURCE >= 700
+    my_stat->st_atim = stat->st_atim;
+    my_stat->st_mtim = stat->st_mtim;
+    my_stat->st_ctim = stat->st_ctim;
+#else
+    my_stat->st_atim = stat->st_atimespec;
```

Comment:
From a quick WWW search `st_atimespec` looks like Apple.
That means that there should probably be an entry
for in (equivalent to st_atim) in `statOps` . 

Yes it is non-POSIX, and should probably be marked as such, but it makes life simpler for macOS users (and us supporting them).

Same Idea for `st_mktimespec` & `st_ctimespec`

I think that I would do a separate section in the class
(and use a_time,etc of the class) & mark it as non-POSIX
rather than interleave.  You are the one writing the code
your choice.

Reply:
I cveated a separate section in `statOps`.

https://github.com/scala-native/scala-native/pull/3049/files#diff-04b347435486592fb4883323b8d6fde17d0a598b7aaf509c6a263da5b5f1c93eR207